src/main/java/com/ldtteam/domumornamentum/block/AbstractPostBlock.java

```diff
@@ -0,0 +1,228 @@
+package com.ldtteam.domumornamentum.block;
+
+import com.ldtteam.domumornamentum.block.interfaces.IDOBlock;
+import com.ldtteam.domumornamentum.block.types.PostType;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.context.BlockPlaceContext;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.LevelAccessor;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.HorizontalDirectionalBlock;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
+import net.minecraft.world.level.block.state.properties.DirectionProperty;
+import net.minecraft.world.level.block.state.properties.EnumProperty;
+import net.minecraft.world.level.material.FluidState;
+import net.minecraft.world.level.material.Fluids;
+import net.minecraft.world.phys.shapes.BooleanOp;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * The abstract class for structurize-added posts.
+ *
+ * Singular fence-like block, one material channel, custom collision, placing alignment with getclickedface() direction
+ */
+public abstract class AbstractPostBlock<B extends AbstractPostBlock<B>> extends HorizontalDirectionalBlock implements IDOBlock<B> {
+
+    public static final EnumProperty<PostType> TYPE = EnumProperty.create("type", PostType.class);
+    public static final DirectionProperty FACING = BlockStateProperties.FACING;
+    public static final BooleanProperty    UPRIGHT = BlockStateProperties.CONDITIONAL;
+    public static final BooleanProperty WATERLOGGED = BlockStateProperties.WATERLOGGED;
+
+    protected static final VoxelShape Y_AXIS_AABB = Block.box(6.0D, 0.0D, 6.0D, 10.D, 16.0D, 10.0D);
+    protected static final VoxelShape Z_AXIS_AABB = Block.box(6.0D, 6.0D, 0.0D, 10.D, 10.0D, 16.0D);
+    protected static final VoxelShape X_AXIS_AABB = Block.box(0.0D, 6.0D, 6.0D, 16.0D, 10.0D, 10.0D);
+    protected static VoxelShape X_DOUBLE_AABB = Shapes.join(
```

Comment:
I hoenstly would avoid overly complex shapes and make them more simple, even if it means they are not fully aligned. Makes clicking them also easier and  more performant, what do you think @marchermans

Reply:
To be fair, it's just two or four boxes and they don't touch, clicking air and hitting the post and vice versa won't feel right, I just wanted the other post(s) but not have a central column of air be their hitbox.  Shooting past them is a feature, not going to be using booleans to try to do anything as complex as stair, or pane, even fence, or lectern might be the heaviest but it does suit the part.  

Also, looked into being able to use a simple model for collision just to get a diagonal slope.  I've seen other deco mods that have it, but not retexturable and cornering slopes.  If so, you could use it in the shingles, or not.  Whatever performs best